authorization-common/src/main/java/com/sequenceiq/authorization/utils/EventAuthorizationUtils.java

```diff
@@ -0,0 +1,99 @@
+package com.sequenceiq.authorization.utils;
+
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.ADMIN_FREEIPA;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.REGISTER_DATABASE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.CUSTOM_CONFIGURATIONS;
+import static com.sequenceiq.authorization.utils.GetAuthzActionTypeProvider.getActionsForResourceType;
+import static com.sequenceiq.cloudbreak.util.NullUtil.throwIfNull;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.authorization.service.CommonPermissionCheckingUtils;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+
+@Component
+public class EventAuthorizationUtils {
+
+    private static final Map<AuthorizationResourceAction, List<AuthorizationResourceAction>> ALTERNATIVES = Map.of(ADMIN_FREEIPA, List.of(REGISTER_DATABASE));
```

Comment:
maybe it is better if we discuss this `ALTERNATIVES` concept, since I do not get it by the code and I am not sure how `ADMIN_FREEIPA` and `REGISTER_DATABASE` correlates